src/service_aggregator.py

```diff
@@ -709,6 +709,13 @@ def add_knowledge_edge(result_message, aux_graph_ids, answer):
     predicate = qedge["predicates"][0]
     # Create a new knowledge edge
     new_edge_id = str(uuid.uuid4())
+    subname = result_message["message"]["knowledge_graph"]["nodes"][qnode_subject].get("name",qnode_subject)
+    obname = result_message["message"]["knowledge_graph"]["nodes"][qnode_object].get("name",qnode_object)
+    new_edge_id = f"{subname}-{predicate}-{obname}"
+    counter = 0
+    while new_edge_id in result_message["message"]["knowledge_graph"]["edges"]:
+        new_edge_id = f"{subname}-{predicate}-{obname}.{counter}"
+        counter += 1
```

Comment:
I also don't think this is going to do anything, except for within ARAGORN. As soon as the message is made into a reasoner-pydantic object again, it will get a new edge ID which is a hash of subject, predicate, object, qualifiers, and source. 

We can still keep it in, if we think it will be useful for debugging within ARAGORN.

Reply:
Oh, thanks for pointing that out - does the same thing happen with aux graph ids?